miniapps/navier/cht.cpp

```diff
@@ -0,0 +1,604 @@
+   Array<int> ess_tdof_list; // this list remains empty for pure Neumann b.c.
+
+   mutable ParBilinearForm *M;
+   ParBilinearForm *K;
+
+   HypreParMatrix Mmat;
+   HypreParMatrix Kmat;
+   HypreParMatrix *T; // T = M + dt K
+   double current_dt;
+
+   mutable CGSolver M_solver;    // Krylov solver for inverting the mass matrix M
+   HypreSmoother M_prec; // Preconditioner for the mass matrix M
+
+   CGSolver T_solver;    // Implicit solver for T = M + dt K
+   HypreSmoother T_prec; // Preconditioner for the implicit solver
+
+   double alpha, kappa, udir;
+
+   mutable Vector z; // auxiliary vector
+
+public:
+   ConductionOperator(ParFiniteElementSpace &f, double alpha, double kappa,
+                      VectorGridFunctionCoefficient adv_gf_c);
+
+   virtual void Mult(const Vector &u, Vector &du_dt) const;
+   /** Solve the Backward-Euler equation: k = f(u + dt*k, t), for the unknown k.
+       This is the only requirement for high-order SDIRK implicit integration.*/
+   virtual void ImplicitSolve(const double dt, const Vector &u, Vector &k);
+
+   /// Update the diffusion BilinearForm K using the given true-dof vector `u`.
+   void SetParameters(VectorGridFunctionCoefficient adv_gf_c);
+
+   virtual ~ConductionOperator();
+};
+
+void VisualizeField(socketstream &sock, const char *vishost, int visport,
+                    ParGridFunction &gf, const char *title,
+                    int x = 0, int y = 0, int w = 400, int h = 400,
+                    bool vec = false);
+
+int main(int argc, char *argv[])
+{
+   // Initialize MPI.
+   int num_procs, myid;
+   MPI_Init(&argc, &argv);
+   MPI_Comm_size(MPI_COMM_WORLD, &num_procs);
+   MPI_Comm_rank(MPI_COMM_WORLD, &myid);
+
+   // Parse command-line options.
+   int lim_meshes = 2; //should be greater than nmeshes
+   Array <const char *> mesh_file_list(lim_meshes);
+   Array <int> np_list(lim_meshes),
+         rs_levels(lim_meshes);
+   rs_levels                 = 0;
+   np_list                   = 1;
+   bool visualization        = true;
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&np_list[0], "-np1", "--np1",
+                  "number of MPI ranks for mesh 1");
+   args.AddOption(&np_list[1], "-np2", "--np2",
+                  "number of MPI ranks for mesh 1");
+   args.AddOption(&rs_levels[0], "-r1", "--refine-serial 1",
+                  "Number of times to refine the mesh 1 uniformly in serial.");
+   args.AddOption(&rs_levels[1], "-r2", "--refine-serial 2",
+                  "Number of times to refine the mesh 2 uniformly in serial.");
+   args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+
+   args.Parse();
+   if (!args.Good())
+   {
+      args.PrintUsage(cout);
+      return 1;
+   }
+   if (myid == 0)
+   {
+      args.PrintOptions(cout);
+   }
+
+   const int nmeshes         = 2;
+   mesh_file_list[0]         = "fluid-cht.mesh";
```

Comment:
These files are relative to where things are being run/installed. So, this will fail to run with in the cmake case with the current options. One option would be to have users pass in the location. Another might be to do something similar to what happens in the meshing miniapp where at least for the makefile symbolic links are created for the necessary mesh files in that directory. I'll try and find the equivalent way to do that in the cmake files.